python/pyarrow/parquet.py

```diff
@@ -432,14 +432,18 @@ class ParquetDatasetPiece(object):
     ----------
     path : str or pathlib.Path
         Path to file in the file system where this piece is located
+    open_file_func : callable
+        Function to use for obtaining file handle to dataset piece
     partition_keys : list of tuples
       [(column name, ordinal index)]
     row_group : int, default None
         Row group to load. By default, reads all row groups
     """
 
-    def __init__(self, path, row_group=None, partition_keys=None):
+    def __init__(self, path, open_file_func=lambda p: open(p, 'rb'),
```

Comment:
Don't store `lambda` functions in attributes but rather use `partial(open, mode='rb')`. If this object is ever pickled, this makes a huge performance difference.

Reply:
noted, thank you